### DISCUSSION TITLE
undo, rollback?

### DISCUSSION THREAD

**Discussion by wuqiuju** [xyflow/xyflow#2079] [Ideas]
Can reactFlow provide callback undo, rollback? Currently the project requires button control.
Thank you,
![image](https://user-images.githubusercontent.com/26135997/163561408-3c1f0685-9c97-4927-b34a-06212f8c2805.png)

**Comment by raarts** [comment]
I don't know if this will help you, but I wrap my state in [useUndo](https://github.com/homerchen19/use-undo).
